### ISSUE TITLE
No ability to exclude associate_public_ip_address from launch_template

### ISSUE THREAD

**Issue by sedan07** [terraform-providers/terraform-provider-aws#5293]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.10.8

### Affected Resource(s)

* `aws_launch_template`

### Terraform Configuration Files

Setting a network interface ID in a launch template and trying to exclude the `associate_public_ip_address` argument. AWS doesn't let you specify the `associatePublicIPAddress` on a defined network interface. 

e.g.
```
resource "aws_launch_template" "test" {
  name_prefix   = "test-"
  image_id      = "${data.aws_ami.coreos.image_id}"
  instance_type = "${var.instance_type}"

  network_interfaces {
    associate_public_ip_address = "null"
    network_interface_id = "${aws_network_interface.eth0.id}"
    device_index = "0"
  }
}
```

or 
```
resource "aws_launch_template" "test" {
  name_prefix   = "test-"
  image_id      = "${data.aws_ami.coreos.image_id}"
  instance_type = "${var.instance_type}"

  network_interfaces {
    network_interface_id = "${aws_network_interface.eth0.id}"
    device_index = "0"
  }
}
```

### Expected Behavior

Terraform would of excluded the `associatePublicIPAddress` setting from the launch template and then AWS would let you launch an instance or autoscaling group using it.

### Actual Behavior

> You must use a valid fully-formed launch template. The associatePublicIPAddress parameter cannot be specified for a network interface with an ID.

### Steps to Reproduce

1. `terraform apply`

**Comment by bflad** [comment]
The fix for this has been merged into master and will release with version 1.30.0 of the AWS provider, likely middle of next week.

**Comment by bflad** [comment]
This has been released in [version 1.30.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1300-august-02-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
